rpmio/rpmlua.c

```diff
@@ -804,8 +881,13 @@ static int rpm_execute(lua_State *L)
 	return pusherror(L, status, NULL);
     if (waitpid(pid, &status, 0) == -1)
 	return pusherror(L, errno, NULL);
-    if (status != 0)
-	return pusherror(L, status, "exit code");
+    if (status != 0) {
+	if (WIFSIGNALED(status)) {
+	    return pusherror(L, WTERMSIG(status), "exit signal");
+	} else {
+	    return pusherror(L, WEXITSTATUS(status), "exit code");
+	}
+    }
 
     return pushresult(L, status);
```

Comment:
Oops, seems like this one should also be fixed to use `WEXITSTATUS(status)` instead :smile:

Reply:
That will only be reached when status is zero, so it doesn't make any difference.

Reply:
We could of course make the zero explicit to make it clearer.

Reply:
Oh right, of course... nevermind then. We'll be refactoring this code anyway.